api/src/main/java/org/openmrs/module/fhirExtension/service/ObsBasedDiagnosticReportService.java

```diff
@@ -71,10 +72,13 @@ public DiagnosticReport create(@Nonnull DiagnosticReport diagnosticReport) {
 			FhirDiagnosticReport fhirDiagnosticReport = obsBasedDiagnosticReportTranslator.toOpenmrsType(diagnosticReport);
 			diagnosticReportObsValidator.validate(fhirDiagnosticReport);
 			diagnosticReportRequestValidator.validate(diagnosticReport);
-			Set<Obs> createdObs = createObs(fhirDiagnosticReport.getResults());
+			Order order = orderUtil.getOrder(diagnosticReport, fhirDiagnosticReport);
+			Set<Obs> diagnosticObs = fhirDiagnosticReport.getResults();
+			orderUtil.updateObsWithOrder(diagnosticObs, order);
+			Set<Obs> createdObs = createObs(diagnosticObs);
 			fhirDiagnosticReport.setResults(createdObs);
 			FhirDiagnosticReport createdFhirDiagnosticReport = fhirDiagnosticReportDao.createOrUpdate(fhirDiagnosticReport);
-			orderUpdateService.updateOrder(diagnosticReport, fhirDiagnosticReport);
+			orderUtil.updateFulFillerStatus(order);
```

Comment:
Traditionally `utils` specifically in most java projects were used for reusable functions (such as date conversion, isEmpty object etc). It seems you chose `OrderUtil` because there is already `OrderService` from openmrs? 

If you need a service layer for responsibility segregation - you would want to challenge on consumption of that services before you start exposing public contracts e.g. in your case `getOrder` is just and intermediate steps for the final `updateObsWithOrder` - so basically you just need one public contract of `OrderService.update` which first calls the openmrs getOrder followed by updateObsWithOrder (both private construct of your new OrderService)

Reply:
another point to consider is that we have too many services abstracting business logic (or just object conversion) and deligating